Gems/ROS2/Code/Include/ROS2/Lidar/LidarRaycasterBus.h

```diff
@@ -98,12 +112,29 @@ namespace ROS2
         //! @param range Ray range in meters.
         virtual void ConfigureRayRange(float range) = 0;
 
+        //! Configures ray minimum travel distance.
+        //! @param range Ray range in meters.
+        virtual void ConfigureMinimumRayRange(float range)
+        {
+            AZ_Assert(false, "This Lidar Implementation does not support minimum ray range configurations!");
+        }
+
         //! Schedules a raycast that originates from the point described by the lidarTransform.
         //! @param lidarTransform Current transform from global to lidar reference frame.
         //! @return Results of the raycast in form of coordinates in 3D space.
         //! The returned vector size can be anything between zero and size of directions. No hits further than distance will be reported.
         virtual AZStd::vector<AZ::Vector3> PerformRaycast(const AZ::Transform& lidarTransform) = 0;
 
+        //! Schedules a raycast that originates from the point described by the lidarTransform.
+        //! @param lidarTransform Current transform from global to lidar reference frame.
+        //! @param flags Used to request different kinds of data returned by raycast query
+        //! @return Results of the raycast in the requested form.
+        virtual RaycastResult PerformRaycastWithFlags(const AZ::Transform& lidarTransform, RaycastResultFlags flags)
```

Comment:
I would suggest replacing this function with void SetRaycastResultFlags(RaycastResultFlags flags). I've started implementing the rgl-gem implementation of the 2D lidar and setting these flags once before raycast calls could both simplify the rgl-gem-side code and improve performance. RGL has an API call for adjusting the raycast result components and modifying it before each raycast could be moderately costly. Additionally this change would simplify the LidarRaycasterBus interface.

Reply:
Those are some valid points, I like it! Let me do the rework